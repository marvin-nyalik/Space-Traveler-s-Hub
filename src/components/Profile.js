import { useSelector } from 'react-redux';
import css from './Profile.module.css';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets.allRockets);
  const reserved = rockets.filter((rocket) => rocket.reserved === true);

  const missions = useSelector((state) => state.missions.missions);
  const joined = missions.filter((mission) => mission.reserved);
  return (
    <>
      <div className={css.container}>
        <div className={css.missions}>
          <p className={css.heading}>My Missions</p>
          { joined.map((mission) => (
            <p className={css.elem} key={mission.id}>{mission.name}</p>
          ))}
        </div>
        <div className={css.rocketsCont}>
          <p className={css.heading}>My Rockets</p>
          { reserved ? reserved.map((rocket) => (<p key={rocket.id} className={css.elem}>{rocket.rocket_name}</p>)) : ''}
        </div>
      </div>
    </>
  );
};

export default Profile;
