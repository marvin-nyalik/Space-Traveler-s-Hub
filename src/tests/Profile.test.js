import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import Profile from '../components/Profile';

const mockStore = configureMockStore([thunk]);
describe('Profile Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      rockets: {
        allRockets: [
          {
            id: 'rocket1',
            rocket_name: 'Rocket 1',
            reserved: true,
          },
          {
            id: 'rocket2',
            rocket_name: 'Rocket 2',
            reserved: true,
          },
        ],
      },
      missions: {
        missions: [
          {
            id: '1',
            name: 'mission 1',
            description: 'mission 1 desc',
            reserved: true,
          },
          {
            id: '2',
            name: 'mission 2',
            description: 'mission 2 desc',
            reserved: true,
          },
        ],
      },
    });
  });

  test('renders Profile component correctly', () => {
    const { getByText } = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(getByText('My Rockets')).toBeInTheDocument();
  });

  it('Matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
