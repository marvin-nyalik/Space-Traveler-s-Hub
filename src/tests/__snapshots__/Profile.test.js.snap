// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component Matches the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="missions"
    >
      <p
        class="heading"
      >
        My Missions
      </p>
      <p
        class="elem"
      >
        mission 1
      </p>
      <p
        class="elem"
      >
        mission 2
      </p>
    </div>
    <div
      class="rocketsCont"
    >
      <p
        class="heading"
      >
        My Rockets
      </p>
      <p
        class="elem"
      >
        Rocket 1
      </p>
      <p
        class="elem"
      >
        Rocket 2
      </p>
    </div>
  </div>
</div>
`;
